iml-gui/crate/src/components/command_modal.rs

```diff
@@ -284,3 +548,638 @@ fn close_button() -> Node<Msg> {
     ]
     .map_msg(Msg::Modal)
 }
+
+async fn fetch_the_batch<T, F, U>(ids: Vec<u32>, data_to_msg: F) -> Result<U, U>
+where
+    T: DeserializeOwned + EndpointName + 'static,
+    F: FnOnce(ResponseDataResult<ApiList<T>>) -> U,
+    U: 'static,
+{
+    // e.g. GET /api/something/?id__in=1&id__in=2&id__in=11&limit=0
+    let err_msg = format!("Bad query for {}: {:?}", T::endpoint_name(), ids);
+    let mut ids: Vec<_> = ids.into_iter().map(|x| ("id__in", x)).collect();
+    ids.push(("limit", 0));
+    Request::api_query(T::endpoint_name(), &ids)
+        .expect(&err_msg)
+        .fetch_json_data(data_to_msg)
+        .await
+}
+
+fn extract_uri_id<T: EndpointName>(input: &str) -> Option<u32> {
+    lazy_static::lazy_static! {
+        static ref RE: Regex = Regex::new(r"/api/(\w+)/(\d+)/").unwrap();
+    }
+    RE.captures(input).and_then(|cap: Captures| {
+        let s = cap.get(1).unwrap().as_str();
+        let t = cap.get(2).unwrap().as_str();
+        if s == T::endpoint_name() {
+            t.parse::<u32>().ok()
```

Comment:
nit:
```suggestion
        if s == T::endpoint_name() {
            let t = cap.get(2).unwrap().as_str();        
            t.parse::<u32>().ok()
```

Reply:
Meh, this visually less nice and the condition always true for all current usages...